src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/SurveySvcRestClientImpl.java

```diff
@@ -123,46 +124,55 @@ private SurveyDTO getSurveyDtoFromResponseEntity(ResponseEntity<String> response
   }
 
   @Override
-  public SurveyDTO findSurvey(UUID surveyId) throws RestClientException {
+  public SurveyDTO findSurvey(final UUID surveyId) throws CTPException {
     UriComponents uriComponents =
         restUtility.createUriComponents(
             appConfig.getSurveySvc().getSurveyDetailPath(), null, surveyId);
 
     HttpEntity<?> httpEntity = restUtility.createHttpEntity(null);
+    ResponseEntity<String> responseEntity = null;
 
     try {
       log.debug(
           "about to get to the Survey SVC with surveyId {} from {}",
           surveyId,
           uriComponents.toUri());
-      ResponseEntity<String> responseEntity =
+      responseEntity =
           restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
-
       return getSurveyDtoFromResponseEntity(responseEntity);
     } catch (RestClientException e) {
-      return null;
+      if (responseEntity != null && responseEntity.getStatusCode().is4xxClientError()) {
```

Comment:
Not sure if this is adding any value. The behaviour spring mvc has is that it will already do this.

Reply:
RestTemplate is not Spring MVC and the previous behaviour was to catch the exception and return null on any RestClientException. 404 is an expected exception where we should return null, but what about any other kind of problems with the REST endpoint we're calling?

It looks to me like we'll lose all error logging and ability to use retry annotations if we just return null whenever there's a problem - we're introducing hard to diagnose flakiness.

Reply:
Sorry meant MVC will catch the RestClient exception with the generic exception handler